plugins/samba.lua

```diff
@@ -0,0 +1,130 @@
+if lldap == nil then
+    lldap = {}
+end
+
+local ensure_user_attribute_exists = function(context, schema, attribute_name, attribute_type)
+    if schema.user_attributes.attributes[attribute_name] == nil then
+        local res = context.api:add_user_attribute({
```

Comment:
It seems like you could just create the attribute regardless and swallow the error.